druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java

```diff
@@ -1368,6 +1368,35 @@ public Void apply(ResultSet resultSet) {
     sql(sql, WIKI)
         .returnsCount(9);
   }
+
+  @Test public void testFieldBasedCost() {
+    String sql = "select \"countryName\", floor(\"time\" to DAY), cast(count(*) as integer) as c\n"
+                 + "         from \"wiki\"\n"
+                 + "         where floor(\"time\" to DAY) >= '1997-01-01 00:00:00' and "
+                 + "         floor(\"time\" to DAY) < '1997-09-01 00:00:00'\n"
+                 + "         group by \"countryName\", floor(\"time\" TO DAY)\n"
+                 + "         order by c limit 5";
+
+    String plan = "BindableProject(countryName=[$0], EXPR$1=[$1], C=[CAST($2):INTEGER NOT NULL])\n"
+                  + "    BindableSort(sort0=[$2], dir0=[ASC], fetch=[5])\n"
+                  + "      BindableAggregate(group=[{0, 1}], agg#0=[COUNT()])\n"
+                  + "        BindableProject(countryName=[$1], EXPR$1=[FLOOR($0, FLAG(DAY))])\n"
+                  + "          BindableFilter(condition=[AND(>=(FLOOR($0, FLAG(DAY)), 1997-01-01 00:00:00), <(FLOOR($0, FLAG(DAY)), 1997-09-01 00:00:00))])\n"
+                  + "            DruidQuery(table=[[wiki, wiki]], intervals=[[1900-01-09T00:00:00.000/2992-01-10T00:00:00.000]], projects=[[$0, $5]])";
+
+    String druidQuery = "{\"queryType\":\"select\","
+                        + "\"dataSource\":\"wikiticker\","
+                        + "\"descending\":false,"
+                        + "\"intervals\":[\"1900-01-09T00:00:00.000/2992-01-10T00:00:00.000\"],"
+                        + "\"dimensions\":[\"countryName\"],"
```

Comment:
@julianhyde @jcamachor: looked at the cost function of Project and TableScan For the above query in question, IMO, the issue is with the cost function for TableScan only takes into account the number of rows in the result set and not the width of the row or the number of fields to be scanned. 
I think for columnar data stores the number of columns that needs to be scanned in a TableScan should be accounted in the cost of a TableScan. Any thoughts on adding that ? 

Also, I think changing the cost of tableScan might be an involved change and should be done in a separate JIRA and we can get include fields in DruidQuery until that is resolved.